resources/js/mp-history/event.tsx

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import TimeWithTooltip from 'components/time-with-tooltip';
+import { route } from 'laroute';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { trans } from 'utils/lang';
+import { linkHtml } from 'utils/url';
+import { MatchEvent } from '../interfaces/match-json';
+import UserJson from '../interfaces/user-json';
+
+interface Props {
+  event: MatchEvent;
+  users: { [userId: number]: UserJson };
+}
+
+const icons = {
+  'host-changed': ['fas fa-exchange-alt'],
+  'match-created': ['fas fa-plus'],
+  'match-disbanded': ['fas fa-times'],
+  'player-joined': ['fas fa-arrow-right', 'far fa-circle'],
+  'player-kicked': ['fas fa-arrow-left', 'fas fa-ban'],
+  'player-left': ['fas fa-arrow-left', 'far fa-circle'],
+};
+
+export default function Event(props: Props) {
+  const user = props.event.user_id != null ? props.users[props.event.user_id] : undefined;
+
+  const event_type = props.event.detail.type;
+  if (event_type === 'other') {
+    return null;
+  }
```

Comment:
Explicitly handled even though it wasn't before because otherwise ts yells at me that I can't use this to index into `icons` later. And none of this basically can handle the `other` type anyway (translations included).